crates/solver/src/liquidity/zeroex.rs

```diff
@@ -68,24 +68,24 @@ impl ZeroExLiquidity {
         allowances: Arc<Allowances>,
     ) -> Option<Liquidity> {
         let sell_amount: U256 = record.remaining_maker_amount().ok()?.into();
-        if sell_amount.is_zero() || record.metadata.remaining_fillable_taker_amount == 0 {
+        if sell_amount.is_zero() || record.metadata().remaining_fillable_taker_amount == 0 {
             // filter out orders with 0 amounts to prevent errors in the solver
             return None;
         }
 
         let limit_order = LimitOrder {
             id: LimitOrderId::Liquidity(LiquidityOrderId::ZeroEx(hex::encode(
-                &record.metadata.order_hash,
+                &record.metadata().order_hash,
             ))),
-            sell_token: record.order.maker_token,
-            buy_token: record.order.taker_token,
+            sell_token: record.order().maker_token,
+            buy_token: record.order().taker_token,
             sell_amount,
-            buy_amount: record.metadata.remaining_fillable_taker_amount.into(),
+            buy_amount: record.metadata().remaining_fillable_taker_amount.into(),
             kind: OrderKind::Buy,
             partially_fillable: true,
             user_fee: U256::zero(),
             settlement_handling: Arc::new(OrderSettlementHandler {
-                order: record.order,
+                order: record.order().clone(),
```

Comment:
Now we should store the entire `OrderRecord` in the `OrderSettlementHandler`.

Reply:
Fixed